### ISSUE TITLE
Suggestions

### ISSUE THREAD

**Issue by dgl-prc** [eth-sri/eran#24]
1. could you please offer an example or a tutorial about how a qualified tensorflow model is trained and saved, as well as the version of tensorflow this project is based on? 

   > There are various problems when I try to verify a model trained by myself. It seems that there is a set of rules to conform to when defining a graph and save a trained model. An example including all phases, i.e., from the definition of a model graph to model training to model saving and finally to model verification, will greatly facilitate learning and using ERAN.

**Comment by GgnDpSngh** [comment]
Hi there,

Frankly, it is a mess to import models directly from TensorFlow. This is because ERAN is designed for Tensorflow version 1.11 or higher. However, version 2 is not compatible with the older version, and supporting two incompatible versions creates a big mess for us (e.g., models trained with version 2 *do not* run with version 1 support in ERAN and vice-versa). The best option would be to export your model into ONNX format which ERAN can handle. You can install the Tensorflow to the ONNX converter from here:
https://github.com/onnx/tensorflow-onnx

The command for the conversion is:
python -m tf2onnx.convert --saved-model tensorflow-model-path --opset 10 --output model.onnx

I am attaching an example code for training a simple fully connected network in Tensorflow taken from https://www.tensorflow.org/datasets/keras_example 

import tensorflow.compat.v2 as tf
import tensorflow_datasets as tfds

tfds.disable_progress_bar()
#tf.enable_v2_behavior()

(ds_train, ds_test), ds_info = tfds.load(
    'mnist',
    split=['train', 'test'],
    shuffle_files=True,
    as_supervised=True,
    with_info=True,
)

def normalize_img(image, label):
  """Normalizes images: `uint8` -> `float32`."""
  return tf.cast(image, tf.float32) / 255., label

ds_train = ds_train.map(
    normalize_img, num_parallel_calls=tf.data.experimental.AUTOTUNE)
ds_train = ds_train.cache()
ds_train = ds_train.shuffle(ds_info.splits['train'].num_examples)
ds_train = ds_train.batch(128)
ds_train = ds_train.prefetch(tf.data.experimental.AUTOTUNE)

ds_test = ds_test.map(
    normalize_img, num_parallel_calls=tf.data.experimental.AUTOTUNE)
ds_test = ds_test.batch(128)
ds_test = ds_test.cache()
ds_test = ds_test.prefetch(tf.data.experimental.AUTOTUNE)

model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128,activation='relu'),
  tf.keras.layers.Dense(10, activation='softmax')
])
model.compile(
    loss='sparse_categorical_crossentropy',
    optimizer=tf.keras.optimizers.Adam(0.001),
    metrics=['accuracy'],
)

model.fit(
    ds_train,
    epochs=6,
    validation_data=ds_test,
)

tf.saved_model.save(model, 'mnist_net')


I have updated the ERAN code so that the Softmax layer is ignored as it does not affect the classification result. Let me know if there are any further issues. 

Cheers,
Gagandeep Singh

**Comment by dgl-prc** [comment]
Thank you so much! I tried many times and finally turned to pytorch+onnx. 
With pytorch 1.5 and onnx 1.7, I successfully verified my toy model :)
